src/mbgl/text/glyph_pbf.cpp

```diff
@@ -64,26 +65,35 @@ void parseGlyphPBF(GlyphSet& glyphSet, const GlyphRange& glyphRange, const std::
                 }
             }
 
+            // Only treat this glyph as a correct glyph if it has all required fields. It also
+            // needs to satisfy a few metrics conditions that ensure that the glyph isn't bogus.
+            // All other glyphs are malformed.  We're also discarding all glyphs that are outside
+            // the expected glyph range.
+            if (!hasID || !hasWidth || !hasHeight || !hasLeft || !hasTop || !hasAdvance ||
+                glyph.metrics.width >= 256 || glyph.metrics.height >= 256 ||
+                glyph.metrics.left < -128 || glyph.metrics.left >= 128 ||
+                glyph.metrics.top < -128 || glyph.metrics.top >= 128 ||
+                glyph.metrics.advance >= 256 ||
+                glyph.id < glyphRange.first || glyph.id > glyphRange.second) {
+                continue;
+            }
+
             // If the area of width/height is non-zero, we need to adjust the expected size
             // with the implicit border size, otherwise we expect there to be no bitmap at all.
-            const uint32_t expectedBitmapSize =
-                glyph.metrics.width && glyph.metrics.height
-                    ? (glyph.metrics.width + 2 * SDFGlyph::borderSize) *
-                          (glyph.metrics.height + 2 * SDFGlyph::borderSize)
-                    : 0;
-
-            // Only treat this glyph as a correct glyph if it has all required fields, and if
-            // the bitmap has the correct length. It also needs to satisfy a few metrics conditions
-            // that ensure that the glyph isn't bogus. All other glyphs are malformed.
-            // We're also discarding all glyphs that are outside the expected glyph range.
-            if (hasID && hasWidth && hasHeight && hasLeft && hasTop && hasAdvance &&
-                glyph.metrics.width < 256 && glyph.metrics.height < 256 &&
-                glyph.metrics.left >= -128 && glyph.metrics.left < 128 &&
-                glyph.metrics.top >= -128 && glyph.metrics.top < 128 &&
-                glyph.metrics.advance < 256 && glyph.bitmap.size() == expectedBitmapSize &&
-                glyph.id >= glyphRange.first && glyph.id <= glyphRange.second) {
-                glyphSet.insert(glyph.id, std::move(glyph));
+            if (glyph.metrics.width && glyph.metrics.height) {
+                const Size size {
+                    glyph.metrics.width + 2 * SDFGlyph::borderSize,
+                    glyph.metrics.height + 2 * SDFGlyph::borderSize
+                };
+
+                if (size.area() != glyphData.size()) {
```

Comment:
The width/height comparison would be `size.width * size.height != glyphData.size()`. Is that what you mean?